cmd/account/show/delegations.go

```diff
@@ -0,0 +1,301 @@
+		if desc.self {
+			fmt.Fprintf(w, " (self)")
+		}
+		fmt.Fprintln(w)
+		fmt.Fprintf(w, "%s    %-*s ", prefix, lenLongest, amountFieldName)
+		fmt.Fprintf(w, "%s", helpers.FormatConsensusDenomination(network, desc.amount))
+		fmt.Fprintf(w, " (%s shares)\n", desc.shares)
+		if desc.endTime != beacon.EpochInvalid {
+			fmt.Fprintf(w, "%s    %-*s epoch %d\n", prefix, lenLongest, endTimeFieldName, desc.endTime)
+		}
+	}
+}
+
+// prettyPrintAccountBalanceAndDelegationsFrom pretty-prints the given account's general balance and
+// (outgoing) delegations from this account.
+func prettyPrintAccountBalanceAndDelegationsFrom(
+	network *config.Network,
+	addr *types.Address,
+	generalAccount staking.GeneralAccount,
+	actDelegationInfos map[staking.Address]*staking.DelegationInfo,
+	debDelegationInfos map[staking.Address][]*staking.DebondingDelegationInfo,
+	prefix string,
+	w io.Writer,
+) {
+	var totalActDelegationsAmount, totalDebDelegationsAmount types.Quantity
+
+	consensusAddr := addr.ConsensusAddress()
+	availableAmount := generalAccount.Balance
+	totalAmount := availableAmount.Clone()
+
+	actDelegationDescs := make([]delegationDescription, 0, len(actDelegationInfos))
+
+	for delAddr, delInfo := range actDelegationInfos {
+		delDesc := delegationDescription{
+			delAddr,
+			delAddr.Equal(consensusAddr),
+			delegationAmount(delInfo.Shares, delInfo.Pool),
+			delInfo.Shares,
+			beacon.EpochInvalid,
+		}
+		actDelegationDescs = append(actDelegationDescs, delDesc)
+		_ = totalActDelegationsAmount.Add(&delDesc.amount)
+	}
+	_ = totalAmount.Add(&totalActDelegationsAmount)
+
+	debDelegationDescs := make([]delegationDescription, 0, len(debDelegationInfos))
+
+	for delAddr, delInfoList := range debDelegationInfos {
+		for _, delInfo := range delInfoList {
+			delDesc := delegationDescription{
+				delAddr,
+				delAddr.Equal(consensusAddr),
+				delegationAmount(delInfo.Shares, delInfo.Pool),
+				delInfo.Shares,
+				delInfo.DebondEndTime,
+			}
+			debDelegationDescs = append(debDelegationDescs, delDesc)
+			_ = totalDebDelegationsAmount.Add(&delDesc.amount)
+		}
+	}
+	_ = totalAmount.Add(&totalDebDelegationsAmount)
+
+	fmt.Fprintf(w, "%sTotal: ", prefix)
+	fmt.Fprintf(w, "%s\n", helpers.FormatConsensusDenomination(network, *totalAmount))
+	fmt.Fprintf(w, "%sAvailable: ", prefix)
+	fmt.Fprintf(w, "%s\n", helpers.FormatConsensusDenomination(network, availableAmount))
+	fmt.Fprintln(w)
+
+	innerPrefix := prefix + "  "
+	const addressFieldName = "To:"
+
+	if len(actDelegationDescs) > 0 {
+		fmt.Fprintf(w, "%sActive Delegations from this Account:\n", prefix)
+		fmt.Fprintf(w, "%sTotal: ", innerPrefix)
+		fmt.Fprintf(w, "%s\n", helpers.FormatConsensusDenomination(network, totalActDelegationsAmount))
+		fmt.Fprintln(w)
+
+		sort.Sort(byEndTimeAmountAddress(actDelegationDescs))
+		prettyPrintDelegationDescriptions(network, actDelegationDescs, addressFieldName, innerPrefix, w)
+		fmt.Fprintln(w)
+	}
+
+	if len(debDelegationDescs) > 0 {
```

Comment:
This does not seem to work as expected. Currently `test:dave` has 200 TEST in debonding period that were delegated to `test:erin`, but showing their accounts doesn't show debonding balance under PARATIME section.

Steps to reproduce:
```
oasis acc delegate 100 test:erin --account test:dave --network testnet --paratime sapphire
oasis acc show test:dave --show-delegations
oasis acc show test:erin --show-delegations
# active delegations correctly printed under the PARATIME section
# but:
oasis acc undelegate 100000000000 test:erin --account test:dave --network testnet --paratime sapphire
oasis acc show test:dave --show-delegations
# ^^ doesn't show Debonding Delegations from this Account section
oasis acc show test:erin --show-delegations
# ^^ shows Debonding Delegations to this Account section, but under consensus!
```

Reply:
FYI delegating/undelegating on Consensus works as expected.

Reply:
Noticed this will be done after https://github.com/oasisprotocol/oasis-sdk/pull/1466 is merged.